src/celeritas/em/model/MuBetheBlochModel.cc

```diff
@@ -0,0 +1,111 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/em/model/MuBetheBlochModel.cc
+//---------------------------------------------------------------------------//
+#include "MuBetheBlochModel.hh"
+
+#include "celeritas/Quantities.hh"
+#include "celeritas/em/data/MuBetheBlochData.hh"
+#include "celeritas/em/executor/MuBetheBlochExecutor.hh"
+#include "celeritas/em/interactor/detail/PhysicsConstants.hh"
+#include "celeritas/global/ActionLauncher.hh"
+#include "celeritas/global/CoreParams.hh"
+#include "celeritas/global/CoreState.hh"
+#include "celeritas/global/TrackExecutor.hh"
+#include "celeritas/phys/InteractionApplier.hh"
+#include "celeritas/phys/PDGNumber.hh"
+#include "celeritas/phys/ParticleParams.hh"
+#include "celeritas/phys/ParticleView.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from model ID and other necessary data.
+ */
+MuBetheBlochModel::MuBetheBlochModel(ActionId id,
+                                     ParticleParams const& particles)
+{
+    CELER_EXPECT(id);
+    data_.ids.action = id;
+    data_.ids.electron = particles.find(pdg::electron());
+    data_.ids.mu_minus = particles.find(pdg::mu_minus());
+    data_.ids.mu_plus = particles.find(pdg::mu_plus());
+
+    CELER_VALIDATE(
+        data_.ids.electron && data_.ids.mu_minus && data_.ids.mu_plus,
+        << "missing electron and/or muon particles (required for "
+        << this->description() << ")");
+
+    data_.electron_mass = particles.get(data_.ids.electron).mass();
```

Comment:
Can we do the same way for muon mass (mass_) instead of setting it from particle.mass() in the constructor or is the latter more safer?

Reply:
I thought we only stored a particle mass in the data when we weren't able to get it from the incident particle (e.g. if we needed electron mass in a gamma model), but I see now there are several exceptions. It's not _necessaary_ to store it in the model data, but is there a benefit to doing that vs. getting it directly from the particle?

Reply:
I guess it's maybe slightly less indirection but more code and more work... let's keep to using the incident `Particle` insofar as possible.

Reply:
I do not have a con or pro for this as they are essentially same.  Getting it directly from the particle input is assuming that the particle is, in this, either mu+ or mu- of which id will be assured later in the body of the constructor (which is of course self-satisfactory each other).  Agree to set a consistent rule for this if @sethrj has a final thought.